packages/api/src/platforms/vtex/utils/skuVariants.ts

```diff
@@ -0,0 +1,168 @@
+import { StoreProduct as StoreProductType } from '../../..'
+import type { Product, Item } from '../clients/search/types/ProductSearchResult'
+
+export type SkuVariants = StoreProductType[]
+
+export type SkuVariantsByName = Record<
+  string,
+  Array<{ alt: string; src: string; label: string; value: string }>
+>
+
+function findSkuVariantImage(availableImages: Item['images']) {
+  return (
+    availableImages.find(
+      (imageProperties) => imageProperties.imageLabel === 'skuvariation'
+    ) ?? availableImages[0]
+  )
+}
+
+export function createSlugsMap(
+  variants: Item[],
+  mainVariant: string,
+  baseSlug: string
+) {
+  /**
+   * Maps property value combinations to their respective SKU's slug. Enables
+   * us to retrieve the slug for the SKU that matches the currently selected
+   * variations in O(1) time.
+   *
+   * Example: `'Color-Red-Size-40': 'classic-shoes-37'`
+   */
+  const slugsMap: Record<string, string> = {}
+
+  variants.forEach((variant) => {
+    const skuSpecificationProperties = variant.variations
+
+    if (skuSpecificationProperties.length === 0) {
+      return
+    }
+
+    // Make sure that the 'name-value' pair for the `mainVariant` variation
+    // is always the first one.
+    let skuVariantKey = `${mainVariant}-${
+      skuSpecificationProperties.find(
+        (variationDetails) => variationDetails.name === mainVariant
+      )?.values[0] ?? ''
+    }`
+
+    skuSpecificationProperties.forEach((property) => {
+      skuVariantKey +=
+        property.name !== mainVariant
+          ? `-${property.name}-${property.values[0]}`
+          : ''
+    })
+
+    slugsMap[skuVariantKey] = `${baseSlug}-${variant.itemId}`
+  })
+
+  return slugsMap
```

Comment:
I agree regarding the innermost `reduce` (to calculate the value for `skuVariant`), but I'm kinda skeptical about the outermost one. I say that because what I understand as the goal of using `reduce` is to iterate over an array and compute a single value based on that original array, in a way that is more in line with the functional programming way.

But I don't think that's exactly what's being done to calculate the `slugsMap`, as for each iteration, we're returning a totally different type from the ones in the input array (`Item` vs `Record<string, string>`), and for each iteration, the computed value from the previous iteration has no impact. Am I making sense? Hahahahah this is kinda tricky to explain, but I do think a regular `for` or `forEach` makes more sense since they're meant to simply iterate over elements in an array, not return anything.

For calculating the keys of the `slugsMap`, I totally agree that a reduce makes a lot of sense! Because we're indeed calculating a single value from the elements of an array, and the result of each iteration builds on the previous one.

Reply:
I agree with you! it makes a lot of sense the innermost, and the outermost is questionable hahaha. 
Let's go ahead!